Lib9c/Model/State/AvatarState.cs

```diff
@@ -65,11 +65,32 @@ public static Address CreateAvatarAddress()
             return key.PublicKey.Address;
         }
 
-        public AvatarState(Address address,
+        public static AvatarState Create(Address address,
             Address agentAddress,
             long blockIndex,
             AvatarSheets avatarSheets,
             Address rankingMapAddress,
+            string name = null)
+        {
+            var worldInformationVar = new WorldInformation(blockIndex, avatarSheets.WorldSheet,
+                GameConfig.IsEditor, name);
+            var questListVar = new QuestList(
+                avatarSheets.QuestSheet,
+                avatarSheets.QuestRewardSheet,
+                avatarSheets.QuestItemRewardSheet,
+                avatarSheets.EquipmentItemRecipeSheet,
+                avatarSheets.EquipmentItemSubRecipeSheet
+            );
+            return new AvatarState(
+                address, agentAddress, blockIndex, questListVar, worldInformationVar, rankingMapAddress, name);
+        }
+
+        public AvatarState(Address address,
```

Comment:
What if we take the constructor private and depend only on "AvatarState.Create"?

Reply:
The purpose of this pull request was to easily create an `AvatarState` instance without using `AvatarSheets`. We can change the constructor's access modifier to `internal` and allow to access to `internal` elements to the `Lib9c.Tests` project ([it already opened](https://github.com/planetarium/lib9c/blob/776c9113f7fd8f99b5645222d45b57dd7ad2cb82/Lib9c/AssemblyInfo.cs#L3C38-L3C43)), but it cannot open to `NIneChronicles.Headless.Tests` and other external test projects.